nemo/collections/cv/parts/preprocessing/features.py

```diff
@@ -0,0 +1,41 @@
+import os
+import tempfile
+
+import numpy as np
+import torchvision
+
+
+class VideoFeaturizer(object):
+    def __init__(self):
+        pass
+
+    def process(self, video_file, offset, duration):
+
+        # Load Video
+        video = self.from_file(video_file, offset=offset, duration=duration)
+
+        return video
+
+    def from_file(self, video_file, offset, duration):
+
+        # Load from filename
+        if isinstance(video_file, str):
+            video, audio, infos = torchvision.io.read_video(
+                video_file, start_pts=offset, end_pts=offset + duration, pts_unit="sec"
+            )
+
+        # Load from bytes
+        elif isinstance(video_file, bytes):
+
+            # webdataset.torch_video
+            with tempfile.TemporaryDirectory() as dirname:
+                fname = os.path.join(dirname, f"file.mp4")
+                with open(fname, "wb") as stream:
+                    stream.write(video_file)
```

Comment:
Can we use something like https://docs.python.org/3/library/io.html#binary-i-o to avoid writing to disk and immediately reading from there?

Reply:
Yes I suppose it should be possible, I tried without success when coding the function. So I ended up using the same way of loading videos as WebDataset